hadoop-ozone/integration-test-s3/src/test/java/org/apache/hadoop/ozone/s3/ProxyServer.java

```diff
@@ -0,0 +1,251 @@
+      String remoteAddr = exchange.getRemoteAddress().getAddress().getHostAddress();
+      String xff = exchange.getRequestHeaders().getFirst("X-Forwarded-For");
+
+      if (StringUtils.isNotBlank(xff)) {
+        conn.setRequestProperty("X-Forwarded-For", xff + ", " + remoteAddr);
+      } else {
+        conn.setRequestProperty("X-Forwarded-For", remoteAddr);
+      }
+
+      conn.setRequestProperty("X-Forwarded-Proto", "http");
+    }
+
+    private void copyResponseBody(HttpExchange exchange, HttpURLConnection conn, int responseCode) throws IOException {
+      String transferEncoding = conn.getHeaderField("Transfer-Encoding");
+      String contentLengthStr = conn.getHeaderField("Content-Length");
+
+      if (transferEncoding != null && transferEncoding.equalsIgnoreCase("chunked")) {
+        exchange.sendResponseHeaders(responseCode, 0);
+      } else {
+        long contentLength = Optional.ofNullable(contentLengthStr).map(Long::parseLong).orElse(0L);
+        exchange.sendResponseHeaders(responseCode, contentLength);
+      }
+
+      try (InputStream is = responseCode >= 400 ? conn.getErrorStream() : conn.getInputStream();
+           OutputStream os = exchange.getResponseBody()) {
+        if (is != null) {
+          IOUtils.copyLarge(is, os, IO_BUFFER.get());
+          os.flush();
+        }
+      }
+    }
+
+    private HttpURLConnection createConnection(HttpExchange exchange, String target) throws IOException {
+      HttpURLConnection conn = (HttpURLConnection) new URL(target).openConnection();
+      String requestMethod = exchange.getRequestMethod();
+      conn.setRequestMethod(requestMethod);
+      conn.setConnectTimeout(CONNECT_TIMEOUT_MS);
+      conn.setReadTimeout(READ_TIMEOUT_MS);
+      return conn;
+    }
+
+    private void copyRequestBody(HttpExchange exchange, HttpURLConnection conn) throws IOException {
+      String method = exchange.getRequestMethod().toLowerCase();
+      if (method.equals("post") || method.equals("put") || method.equals("patch") || method.equals("delete")) {
+        String contentLength = exchange.getRequestHeaders().getFirst("Content-Length");
+        conn.setDoOutput(true);
+        if (StringUtils.isNotBlank(contentLength)) {
+          conn.setFixedLengthStreamingMode(Long.parseLong(contentLength));
+        } else {
+          conn.setChunkedStreamingMode(0);
+        }
+        try (InputStream reqBody = exchange.getRequestBody();
+             OutputStream os = conn.getOutputStream()) {
+          IOUtils.copy(reqBody, os);
+        }
+      }
+    }
+
+    private void copyRequestHeaders(HttpExchange exchange, HttpURLConnection conn) {
+      for (String headerName : exchange.getRequestHeaders().keySet()) {
+        if (isHopByHopHeader(headerName)) {
+          continue;
+        }
+
+        // Skip Expect header to avoid 100-continue issue
+        if ("Expect".equalsIgnoreCase(headerName)) {
+          continue;
+        }
```

Comment:
Any issue if this is removed? From my research in https://issues.apache.org/jira/browse/HDDS-12842, it seems 100-Continue is handled automatically by S3G Jetty server.

Reply:
@hevinhsu I see, good to know. My understanding is the 100-Continue contains empty response body and should be acknowledged by 100 Continue response. If S3G handles it normally, I imagine it might cause the S3G to upload an empty object.

However, I'm fine if this is working with the existing `PutObject` request.